tests/std/tests/P0896R4_views_elements/test.cpp

```diff
@@ -0,0 +1,393 @@
+            if (!is_empty) {
+                assert(*i2 == *i);
+            }
+        }
+
+        if constexpr (CanBegin<const R&>) {
+            const same_as<iterator_t<const R>> auto i3 = as_const(r).begin();
+            if (!is_empty) {
+                assert(*i3 == *i);
+            }
+        }
+    }
+
+    // Validate elements_view::end
+    STATIC_ASSERT(CanMemberEnd<R>);
+    STATIC_ASSERT(CanEnd<const R&> == range<const V>);
+    if (!is_empty) {
+        same_as<sentinel_t<R>> auto i = r.end();
+        static_assert(common_range<R> == common_range<V>);
+        if constexpr (bidirectional_range<R> && common_range<R>) {
+            assert(*prev(i) == *prev(end(expected_keys)));
+        }
+
+        if constexpr (CanEnd<const R&>) {
+            same_as<sentinel_t<const R>> auto i2 = as_const(r).end();
+            static_assert(common_range<const R> == common_range<const V>);
+            if constexpr (bidirectional_range<const R> && common_range<const R>) {
+                assert(*prev(i2) == *prev(end(expected_keys)));
+            }
+        }
+    }
+
+    // Validate view_interface::data
+    STATIC_ASSERT(!CanData<R>);
+    STATIC_ASSERT(!CanData<const R>);
+
+    // Validate view_interface::front and back
+    if (!is_empty) {
+        STATIC_ASSERT(CanMemberFront<R> == forward_range<V>);
+        if constexpr (CanMemberFront<R>) {
+            assert(r.front() == *begin(expected_keys));
+        }
+
+        STATIC_ASSERT(CanMemberBack<R> == (bidirectional_range<V> && common_range<V>) );
+        if constexpr (CanMemberBack<R>) {
+            assert(r.back() == *prev(end(expected_keys)));
+        }
+
+        STATIC_ASSERT(CanMemberFront<const R> == (forward_range<const V>) );
+        if constexpr (CanMemberFront<const R>) {
+            assert(as_const(r).front() == *begin(expected_keys));
+        }
+
+        STATIC_ASSERT(CanMemberBack<const R> == (bidirectional_range<const V> && common_range<const V>) );
+        if constexpr (CanMemberBack<const R>) {
+            assert(as_const(r).back() == *prev(end(expected_keys)));
+        }
+    }
+
+    // Validate view_interface::operator[]
+    if (!is_empty) {
+        STATIC_ASSERT(CanIndex<R> == random_access_range<V>);
+        if constexpr (CanIndex<R>) {
+            assert(r[0] == *r.begin());
+        }
+
+        STATIC_ASSERT(CanIndex<const R> == random_access_range<const V>);
+        if constexpr (CanIndex<const R>) {
+            assert(as_const(r)[0] == *as_const(r).begin());
+        }
+    }
+
+    // Validate elements_view::base() const&
+    STATIC_ASSERT(CanMemberBase<const R&> == copy_constructible<V>);
+    if constexpr (copy_constructible<V> && forward_range<V>) {
+        same_as<V> auto b1 = as_const(r).base();
+        STATIC_ASSERT(noexcept(as_const(r).base()) == is_nothrow_copy_constructible_v<V>);
+        if (!is_empty) {
+            assert(*b1.begin() == pair(0, -1)); // NB: depends on the test data

```

Comment:
Style: Would prefer to see braces for temporary pairs, to make them look less like function calls. Occurs 4 times total.